enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+possible. When bootstrapping completes and node0 reboots, progress
+monitoring will need to shift to the OpenShift APIs (`ClusterVersion`
+and `ClusterOperator`). We will extend the `openshift-install wait-for
+install-complete` command to recognize when the cluster being deployed
+uses the agent installer and to use both APIs for monitoring and
+[progress
+reporting](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/installation-progress-bar.md)
+in that case, including reporting when steps like validation fail and
+the cluster deployment is never going to succeed. Users will still be
+able to write their own monitoring logic using the assisted service
+and OpenShift APIs directly, but they will have the option of using
+the command line tool.
+
+For the fully automated workflow, the steps a user might normally
+perform such as approving hosts or triggering the installation will be
+handled by a new client running as a service on node0. The client will
+read the user's inputs from files included in the image and interact
+with the assisted-service REST API on localhost to drive the
+deployment. For the purposes of this document, we will refer to that
+service as a single entity called `create-cluster`, although in
+practice we may divide some of the steps into separate tools.
+
+### Actors in the Workflow
+
+**data center admin** is a human user responsible for infrastructure
+within the data center, such as an image registry, networking, etc.
+
+**cluster admin** is a human user responsible for deploying a cluster.
+
+**orchestration tool** is intended to be a software component, but the
+steps may also be performed by **cluster admin** if no software
+component is available.
+
+**installer** is the OpenShift installer binary, `openshift-install`.
+
+**node0** is the first host to be added to a cluster.
+
+### Steps for Deploying a Disconnected Cluster
+
+1. The data center admin launches an image registry visible to the
+   hosts that will form the cluster and to a host that has internet
+   access so the release image can be downloaded.
+2. The cluster admin obtains the `oc` command line tool and uses `oc`
+   to copy the OpenShift release image into the local image registry
+   in one of the usual ways with [`oc adm
+   mirror`](https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-installation-images.html)
+   or [the `oc mirror`
+   plugin](https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-disconnected.html).
+3. The cluster admin obtains the OpenShift installer in one of [the
+   usual
+   ways](https://docs.openshift.com/container-platform/4.9/installing/installing_bare_metal/installing-bare-metal.html#installation-obtaining-installer_installing-bare-metal).
+4. The cluster admin triggers their orchestration tool to run the
+   installer.
+5. The orchestration tool prepares an `install-config.yaml` and
```

Comment:
I am leery of using install-config.yaml as the input to `create image`. I assume that many of the fields are irrelevant. Re-using the same file will put a burden on the user to understand which fields are meaningful in which context. It is already cumbersome to have some fields that are only meaningful in the IPI flow, where the documentation describing that they are ignored when doing a UPI install is lacking.

Reply:
Let's take the "none" platform as an example. It is mentioned above that we want to provide the API VIP. That is not currently an option when using the "none" platform. So, would we create a field for the "none" platform where we do not have one now? Would we add a field to that new type for API VIP that would only apply to agent-based installs? How do you convey well to the user that the "none" platform field is only used when doing an agent-based install?

As for fields that would not apply to an agent-based install, let's look at the "baremetal" platform. The following seem like fields that are unlikely to be applicable--which are almost all of the fields in the baremetal field.
libvirtURI
clusterProvisioningIP
provisioningHostIP
bootstrapProvisioningIP
externalBridge
externalMACAddress
provisioningNetwork
provisioningBridge
provisioningMACAddress
provisioningNetworkInterface
provisioningNetworkCIDR
provisioningDHCPExternal
provisioningDHCPRange
bootstrapOSImage
clusterOSImage

Reply:
I think we agreed offline that we would use a separate file for fields needed only for the agent workflow and not for other workflows.